docs/guide/auto_mode.rst

```diff
@@ -0,0 +1,461 @@
+
+.. code:: python
+
+    vehicle.commands.next=2
+    print "Current Waypoint: %s" % vehicle.commands.next
+    vehicle.commands.next=4
+    print "Current Waypoint: %s" % vehicle.commands.next
+
+There is no need to ``flush()`` changes to ``next`` to the vehicle (and as with other attributes, if you fetch a value, it is updated
+from the vehicle).	
+
+
+.. _auto_mode_handle_mission_end: 
+
+Handling the end of a mission
+===============================
+
+At the end of the mission the vehicle will typically "loiter" (hover in place for Copter, 
+circle for Plane, stop for Rover). You can add new commands to the mission, but you will need to toggle from/back to
+AUTO mode to start it running again.
+
+Currently there is no notification in DroneKit when a mission completes. If you need to detect mission end (in order
+to perform some other operation) then you can either:
+
+* Add a dummy mission command and poll :py:func:`Vehicle.commands.next <droneapi.lib.CommandSequence.next>` for the 
+  transition to the final command, or
+* Compare the current position to the position in the last command.
+	
+
+
+
+.. _auto_mode_useful_functions: 
+
+Useful Mission functions
+========================
+
+This example code contains a number of functions that might be useful for managing and monitoring missions:
+
+.. _auto_mode_load_mission_file: 
+
+Load a mission from a file
+-----------------------------
+
+``upload_mission()`` uploads a mission from a file. 
+
+The implementation calls ``readmission()`` (below) to import the mission from a file into a list. The method then
+clears the existing mission and uploads the new version. 
+
+Adding mission commands is discussed :ref:`here in the guide <auto_mode_adding_command>`.
+  
+.. code:: python
+		
+    def upload_mission(aFileName):
+        """
+        Upload a mission from a file.
+        """
+        missionlist = readmission(aFileName)
+        #clear existing mission
+        print 'Clear mission'
+        cmds = vehicle.commands
+        cmds.download()
+        cmds.wait_valid()
+        cmds.clear()
+        vehicle.flush()
+        print 'ClearCount: %s' % cmds.count
+    #add new mission
+    cmds.download()
+    cmds.wait_valid()
+    for command in missionlist:
+        cmds.add(command)
+    vehicle.flush()	
+
+	
+``readmission()`` reads a mission from the specified file and returns a list of :py:class:`Command <droneapi.lib.Command>` objects. 
+
+Each line is split up. The first line is used to test whether the file has the correct (stated) format. 
+For subsequent lines the values are stored in a :py:class:`Command <droneapi.lib.Command>` object 
+(the values are first cast to the correct ``float`` and ``int`` types for their associated parameters).
+The commands are added to a list which is returned by the function.
+  
+.. code:: python
+
+    def readmission(aFileName):
+        """
+        Load a mission from a file into a list.
+	
+    	This function is used by upload_mission().
```

Comment:
Fixed. FYI This sort of crap caused because notepad++ sometimes adds tabs rather than 4 spaces.